lightning-invoice/src/payment.rs

```diff
@@ -1397,18 +1543,203 @@ mod tests {
+		let event_handler = |_: &_| { *event_handled.borrow_mut() = true; };
+
+		let payment_preimage = PaymentPreimage([1; 32]);
+		let payment_invoice = invoice(payment_preimage);
+		let payment_hash = Some(PaymentHash(payment_invoice.payment_hash().clone().into_inner()));
+		let final_value_msat = payment_invoice.amount_milli_satoshis().unwrap();
+
+		let payer = TestPayer::new()
+			.expect_send(Amount::ForInvoice(128))
+			.expect_send(Amount::ForInvoice(128));
+		let final_value_msat = payment_invoice.amount_milli_satoshis().unwrap();
+		let route = TestRouter::route_for_value(final_value_msat);
+		let router = TestRouter {};
+		let scorer = RefCell::new(TestScorer::new()
+			// 1st invoice, 1st path
+	  		.expect_usage(ChannelUsage { amount_msat: 10, inflight_htlc_msat: 0, effective_capacity: EffectiveCapacity::Unknown } )
+			.expect_usage(ChannelUsage { amount_msat: 20, inflight_htlc_msat: 0, effective_capacity: EffectiveCapacity::Unknown } )
+			.expect_usage(ChannelUsage { amount_msat: 64, inflight_htlc_msat: 0, effective_capacity: EffectiveCapacity::Unknown } )
+			// 1st invoice, 2nd path
+			.expect_usage(ChannelUsage { amount_msat: 64, inflight_htlc_msat: 0, effective_capacity: EffectiveCapacity::Unknown } )
+			// 2nd invoice, 1st path
+			.expect_usage(ChannelUsage { amount_msat: 10, inflight_htlc_msat: 0, effective_capacity: EffectiveCapacity::Unknown } )
+		    .expect_usage(ChannelUsage { amount_msat: 20, inflight_htlc_msat: 64, effective_capacity: EffectiveCapacity::Unknown } )
+		    .expect_usage(ChannelUsage { amount_msat: 64, inflight_htlc_msat: 0, effective_capacity: EffectiveCapacity::Unknown } )
+			// 2nd invoice, 2nd path
+			.expect_usage(ChannelUsage { amount_msat: 64, inflight_htlc_msat: 64, effective_capacity: EffectiveCapacity::Unknown } )
+		);
+		let logger = TestLogger::new();
+		let invoice_payer =
+			InvoicePayer::new(&payer, router, &scorer, &logger, event_handler, Retry::Attempts(0));
+
+		// Succeed 1st path, leave 2nd path inflight
+		let payment_id = invoice_payer.pay_invoice(&payment_invoice).unwrap();
+		invoice_payer.handle_event(&Event::PaymentPathSuccessful {
+			payment_id, payment_hash, path: route.paths[0].clone()
+		});
+
+		// Let's pay a second invoice that will be using the same path. This should trigger the
+		// assertions that expect the last 4 ChannelUsage values above where TestScorer is initialized.
+		// Particularly, the 2nd path of the 1st payment, since it is not yet complete, should still
+		// have 64 msats inflight for paths considering the channel with scid of 1.
+		let payment_preimage_2 = PaymentPreimage([2; 32]);
+		let payment_invoice_2 = invoice(payment_preimage_2);
+		invoice_payer.pay_invoice(&payment_invoice_2).unwrap();
+	}
+
+	// This following partial failure tests are variants of `inflight_map_data_trivial_test` above,
+	// except with partial failure errors.
+	#[test]
+	fn partial_failure_with_monitor_update_failed_and_channel_unavailable() {
```

Comment:
Should be named in terms of inflight htlcs. Also, the specific type of error (e.g., channel unavailable) is unimportant so doesn't need to be in the name. Maybe something like `accounts_for_inflight_htlcs_sent_during_partial_failure`.